File: app/src/main/java/com/subhrajyoti/wallify/gallery/FullscreenDialog.java
package com.subhrajyoti.wallify.gallery;

import android.content.Context;
import android.net.Uri;
import android.os.Bundle;
import android.support.v4.app.DialogFragment;
import android.support.v4.view.PagerAdapter;
import android.support.v4.view.ViewPager;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.subhrajyoti.wallify.R;
import com.subhrajyoti.wallify.model.Image;

import java.io.File;
import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;


public class FullscreenDialog extends DialogFragment {
    @BindView(R.id.viewPager)
    ViewPager viewPager;
    @BindView(R.id.countView)
    TextView countView;
    private String TAG = FullscreenDialog.class.getSimpleName();
    private ArrayList<Image> images;
    ViewPager.OnPageChangeListener viewPagerPageChangeListener = new ViewPager.OnPageChangeListener() {

        @Override
        public void onPageSelected(int position) {
            displayMetaInfo(position);
        }

        @Override
        public void onPageScrolled(int arg0, float arg1, int arg2) {

        }

        @Override
        public void onPageScrollStateChanged(int arg0) {

        }
    };
    private Unbinder unbinder;
    private int selectedPosition = 0;

    static FullscreenDialog newInstance() {
        return new FullscreenDialog();
    }

    @SuppressWarnings("unchecked")
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_image_slider, container, false);

        images = (ArrayList<Image>) getArguments().getSerializable("images");
        selectedPosition = getArguments().getInt("position");

        Log.e(TAG, "position: " + selectedPosition);
        Log.e(TAG, "images size: " + images.size());

        unbinder = ButterKnife.bind(this, v);

        MyViewPagerAdapter viewPagerAdapter = new MyViewPagerAdapter();
        viewPager.setAdapter(viewPagerAdapter);
        viewPager.addOnPageChangeListener(viewPagerPageChangeListener);

        setCurrentItem(selectedPosition);

        return v;
    }

    private void setCurrentItem(int position) {
        viewPager.setCurrentItem(position, false);
        displayMetaInfo(selectedPosition);
    }

    private void displayMetaInfo(int position) {
        countView.setText((position + 1) + " of " + images.size());
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setStyle(DialogFragment.STYLE_NORMAL, android.R.style.Theme_Black_NoTitleBar_Fullscreen);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }

    public class MyViewPagerAdapter extends PagerAdapter {

        @BindView(R.id.imageView)
        ImageView imageView;
        private LayoutInflater layoutInflater;

        MyViewPagerAdapter() {
        }

        @Override
        public Object instantiateItem(ViewGroup container, int position) {

            layoutInflater = (LayoutInflater) getActivity().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            View view = layoutInflater.inflate(R.layout.image_fullscreen_preview, container, false);

            Image image = images.get(position);

            ButterKnife.bind(this, view);

            imageView.setImageURI(Uri.fromFile(new File(image.getPath())));

            container.addView(view);

            return view;
        }

        @Override
        public int getCount() {
            return images.size();
        }

        @Override
        public boolean isViewFromObject(View view, Object obj) {
            return view == obj;
        }


        @Override
        public void destroyItem(ViewGroup container, int position, Object object) {
            container.removeView((View) object);
        }
    }
}

File: app/src/main/java/com/subhrajyoti/wallify/gallery/FullscreenDialog.kt
package com.subhrajyoti.wallify.gallery

import android.R.style
import android.content.Context
import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.DialogFragment
import androidx.viewpager.widget.PagerAdapter
import androidx.viewpager.widget.ViewPager.OnPageChangeListener
import com.subhrajyoti.wallify.databinding.FragmentImageSliderBinding
import com.subhrajyoti.wallify.databinding.ImageFullscreenPreviewBinding
import java.io.File
import java.util.ArrayList

class FullscreenDialog : DialogFragment() {

    private var _binding: FragmentImageSliderBinding? = null
    private val binding get() = _binding!!

    private var images: ArrayList<String>? = null
    var viewPagerPageChangeListener: OnPageChangeListener = object : OnPageChangeListener {
        override fun onPageSelected(position: Int) {
            displayMetaInfo(position)
        }

        override fun onPageScrolled(arg0: Int, arg1: Float, arg2: Int) {}
        override fun onPageScrollStateChanged(arg0: Int) {}
    }
    private var selectedPosition = 0
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        _binding = FragmentImageSliderBinding.inflate(inflater)

        images = arguments!!.getSerializable("images") as ArrayList<String>
        selectedPosition = arguments!!.getInt("position")
        val viewPagerAdapter = MyViewPagerAdapter()
        binding.viewPager.adapter = viewPagerAdapter
        binding.viewPager.addOnPageChangeListener(viewPagerPageChangeListener)
        setCurrentItem(selectedPosition)
        return binding.root
    }

    private fun setCurrentItem(position: Int) {
        binding.viewPager.setCurrentItem(position, false)
        displayMetaInfo(selectedPosition)
    }

    private fun displayMetaInfo(position: Int) {
        binding.countView.text = (position + 1).toString() + " of " + images!!.size
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setStyle(STYLE_NORMAL, style.Theme_Black_NoTitleBar_Fullscreen)
    }

    inner class MyViewPagerAdapter : PagerAdapter() {

        override fun instantiateItem(container: ViewGroup, position: Int): Any {
            val layoutInflater = activity!!.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            val binding = ImageFullscreenPreviewBinding.inflate(layoutInflater)
            binding.imageView.setImageURI(Uri.fromFile(File(images!![position])))
            container.addView(binding.root)
            return binding.root
        }

        override fun getCount(): Int {
            return images!!.size
        }

        override fun isViewFromObject(view: View, obj: Any): Boolean {
            return view === obj
        }

        override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
            container.removeView(`object` as View)
        }
    }

    companion object {
        fun newInstance(): FullscreenDialog {
            return FullscreenDialog()
        }
    }
}